test/DynamicSale.test.ts

```diff
@@ -73,7 +74,6 @@ describe('DynamicSale', () => {
 
     it('starts from last price and increments price', async () => {
```

Comment:
This test is even more important that calculate cost.
Running this test you verify that the price is correct (calculateCost) and buy are working properly.
```js
CSV.reduce(async (context, expected) => {
    await Contract.buy('100')
    expect(await USD.balanceOf(buyer)).to.eq([calculation]);
    expect(await TDF.balanceOf(buyer)).to.eq([calculation]);
    // expect minted tokens
    // etc
}, {buyerUSDBalance: parseEther('100000'), buyerTokenBalance: parseEther('0')})
```

This is pseudo code you should take it with a grain of salt